python/hsfs/training_dataset.py

```diff
@@ -325,6 +331,33 @@ def update_statistics_config(self):
         self._training_dataset_engine.update_statistics_config(self)
         return self
 
+    # TODO (davit): write docs
+    def attach_transformation_function(self, builder_function, function_type):
+        self._training_dataset_engine.attach_transformation_function(
+            self, builder_function, function_type
+        )
+        self._transformation_function = (
+            self._training_dataset_engine.get_transformation_function(self)
+        )
+
+    # TODO (davit): write docs
+    def pyspark_transformation(self, builder_function):
+        self._training_dataset_engine.attach_transformation_function(
+            self, builder_function, "pyspark"
+        )
+        self._transformation_function = (
+            self._training_dataset_engine.get_transformation_function(self)
+        )
+
+    # TODO (davit): write docs
+    def python_transformation(self, builder_function):
+        self._training_dataset_engine.attach_transformation_function(
+            self, builder_function, "python"
+        )
+        self._transformation_function = (
+            self._training_dataset_engine.get_transformation_function(self)
+        )
```

Comment:
I think these methods are just confusing. I think we should have a single method `attach_transformation_function`. `function_type` should have a default value.

Why do we need to differentiate between python and pyspark?